cgi-bin/LJ/Protocol.pm

```diff
@@ -1757,6 +1754,11 @@ sub postevent
         # TODO: error on failure?  depends on the job I suppose?  property of the job?
     }
 
+    # To minimize impact on legacy code, let's make sure the entry object in
+    # memory has been populated with data. Easiest way to do that is to call
+    # one of the methods that loads the relevant row from the database.
+    $entry->valid;
```

Comment:
I'll merge it but this feels vaguely magical, which it is, and like we might get bitten at some other point. I like the fix of switching from reaching into the object's members to using the accessor methods. Maybe one step further is having the accessor methods know how to 'instantiate' the right internal data if they're called for the first time and it hasn't been done yet? Does that make sense?